action.yml

```diff
@@ -44,11 +44,12 @@ inputs:
 runs:
   using: 'composite'
   steps:
-    - uses: actions/checkout@v2
+    - name: checkout if not already done
+      if: hashFiles('package.json') == ''
+      uses: actions/checkout@v2.4.0
```

Comment:
I don't think we should be implementing this kind of implicit behavior. If a checkout is needed, we need to decide if we do it inside the action or if we need the user to do it before invoking the action. Having conditional behavior will lead to confusion and bugs most likely.